app/models/lookup_value.rb

```diff
@@ -53,13 +55,17 @@ def validate_and_cast_value
   end
 
   def validate_regexp
-    return true unless (lookup_key.validator_type == 'regexp')
-    errors.add(:value, _("is invalid")) and return false unless (value =~ /#{lookup_key.validator_rule}/)
+    return true if (lookup_key.validator_type != 'regexp' || (lookup_key.contains_erb?(value) && Setting[:interpolate_erb_in_parameters]))
```

Comment:
I definitely agree that the duplication is an issue here but I'd like to keep this patch minimal so it can be easily backported to 1.7. Would creating an issue for refactoring work for you? 

I don't think it's a good idea to use Setting in class context since when you later change the value, the class is not reloaded (in production) so you'd have to restart server in order to apply the change.